### PR TITLE
[map][indexer] Add recycling types to subtitle.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

**Comment by tatiana-yan** [comment]
JTALL

File: indexer/feature_utils.cpp [hunk 1]
```diff
@@ -153,6 +154,36 @@ void GetReadableNameImpl(feature::RegionData const & regionData, StringUtf8Multi
 
   GetMwmLangName(regionData, src, out);
 }
+
+// Filters types with |checker|, returns vector of raw type second components.
+// For example for types {"cuisine-sushi", "cuisine-pizza", "cuisine-seafod"} vector
```
Comment (mpimenov) [inline indexer/feature_utils.cpp:159]: s/seafod/seafood/

File: indexer/feature_utils.cpp [hunk 2]
```diff
@@ -153,6 +154,36 @@ void GetReadableNameImpl(feature::RegionData const & regionData, StringUtf8Multi
...
+  vector<string> localized;
+  for (auto const t : types)
+  {
+    if (!checker(t))
+      continue;
+    localized.push_back(platform::GetLocalizedTypeName(classif().GetReadableObjectName(t)));
+  }
+  return localized;
+}
+
```
Comment (mpimenov) [inline indexer/feature_utils.cpp:186]: лишняя

File: indexer/map_object.hpp [hunk 3]
```diff
@@ -83,6 +83,10 @@ class MapObject
   std::vector<std::string> GetCuisines() const;
   /// @returns translated cuisine(s).
   std::vector<std::string> GetLocalizedCuisines() const;
+  /// @returns not localized recycling type(s).
```
Comment (mpimenov) [inline indexer/map_object.hpp:86]: nit: s/not/non-/ here and for cuisines

**Comment by tatiana-yan** [comment]
JTLINUX

File: indexer/feature_utils.cpp (hunk 1 above)
Comment (tatiana-yan) [reply indexer/feature_utils.cpp]: done

File: indexer/feature_utils.cpp (hunk 2 above)
Comment (tatiana-yan) [reply indexer/feature_utils.cpp]: done

File: indexer/map_object.hpp (hunk 3 above)
Comment (tatiana-yan) [reply indexer/map_object.hpp]: done

**Comment by tatiana-yan** [comment]
и коммит в тесты генератора докинула

File: indexer/map_object.hpp [hunk 4]
```diff
@@ -79,10 +79,14 @@ class MapObject
   std::string GetEmail() const;
   std::string GetWebsite() const;
   Internet GetInternet() const;
-  /// @returns not localized cuisines keys.
+  /// @returns not-localized cuisines keys.
```
Comment (mpimenov) [inline indexer/map_object.hpp:82]: s/not/non/

File: generator/generator_tests/osm_type_test.cpp [hunk 5]
```diff
@@ -1686,6 +1685,9 @@ UNIT_CLASS_TEST(TestWithClassificator, OsmType_ComplexTypesSmoke)
     {{"amenity", "place_of_worship", "muslim"}, {{"amenity", "place_of_worship"}, {"religion", "muslim"}}},
     {{"amenity", "place_of_worship", "shinto"}, {{"amenity", "place_of_worship"}, {"religion", "shinto"}}},
     {{"amenity", "place_of_worship", "taoist"}, {{"amenity", "place_of_worship"}, {"religion", "taoist"}}},
+    {{"amenity", "recycling"}, {{"amenity", "recycling"},{"recycling_type","centre"}}},
```
Comment (mpimenov) [inline generator/generator_tests/osm_type_test.cpp:1688]: clang-format

Comment (tatiana-yan) [reply generator/generator_tests/osm_type_test.cpp]: done

**Comment by tatiana-yan** [comment]
JTLINUX

File: indexer/map_object.hpp (hunk 4 above)
Comment (tatiana-yan) [reply indexer/map_object.hpp]: done

**Comment by tatiana-yan** [comment]
JTLINUX

**Pull request merged**